pkg/cmd/server/api/validation/oauth.go

```diff
@@ -96,18 +96,41 @@ func ValidateIdentityProvider(identityProvider api.IdentityProvider) fielderrors
 		case (*api.HTPasswdPasswordIdentityProvider):
 			allErrs = append(allErrs, ValidateFile(provider.File, "provider.file")...)
 
-		case (*api.OAuthRedirectingIdentityProvider):
-			if len(provider.ClientID) == 0 {
-				allErrs = append(allErrs, fielderrors.NewFieldRequired("provider.clientID"))
+		case (*api.GitHubIdentityProvider):
+			allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
+
+		case (*api.GoogleIdentityProvider):
+			allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
+
+		case (*api.OpenIDIdentityProvider):
```

Comment:
we use standard claims ("sub" is id, "name" is display name, "email" is email address, "preferred_username" is preferred username) unless they override them